azure/templates/post-deploy.yml

```diff
@@ -0,0 +1,109 @@
+parameters:
+  - name: aws_dev
+    default: aws --profile=apim-dev
+
+steps:
+  - bash: |
+      make install-python
+
+    workingDirectory: $(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)
+    displayName: Setup pytests
+    condition: always()
+
+  - template: "azure/components/aws-assume-role.yml@common"
+    parameters:
+        role: "auto-ops"
+        profile: "apim-dev"
+        aws_account: "dev"
+
+  - bash: |
+        echo "We run terraform when it's not a PR release. We also don't run it for any sandbox environment"
+
+        isItPR=0
+        if ! [[ $APIGEE_ENVIRONMENT =~ .*-*sandbox ]]; then
+            export AWS_PROFILE=apim-dev
+            name=$(FULLY_QUALIFIED_SERVICE_NAME)
+            workspace=$(echo $name | grep -oE '[0-9]+$')
+            echo Workspace name is ${workspace}
+
+            if [ -z ${workspace} ]
+            then
+                workspace=$APIGEE_ENVIRONMENT
+            else
+                isItPR=1
+                workspace=pr-${workspace}
+            fi
+
+            echo "##vso[task.setvariable variable=IS_IT_PR]${isItPR}"
+            echo "##vso[task.setvariable variable=WORKSPACE]${workspace}"
+            aws_account_no="$(aws sts get-caller-identity --query Account --output text)"
+            make init
+            echo "Init successful"         
+            make apply aws_account_no=${aws_account_no} environment=${workspace}
+            
+        fi
+    displayName: Run Terraform
+    workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)/terraform"
+    retryCountOnTaskFailure: 2
+
+  - bash: |
+
+      if ! [[ $APIGEE_ENVIRONMENT =~ .*-*sandbox ]]; then
+          echo "Its not sandbox"
+          if [ $(IS_IT_PR) -eq 1 ]
+          then
+              echo "Its PR"
+              #if its not sandbox and it should be PR  
+              make add-endpoint environment=$(WORKSPACE)
+          fi
+
+      fi
+
+    displayName: Add PR-Domain as KeyValue
+    workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)/terraform"
+    retryCountOnTaskFailure: 2
+
+  - bash: |
```

Comment:
Can we combine this step with the one above it? Seems like there's a lot of repeated code here